### DISCUSSION TITLE
Bar Chart with non-time dimension on the X-axis

### DISCUSSION THREAD

**Discussion by tobias-graf-p** [datalust/seq-tickets#1843] [Help] [answered]
I would like to display a bar chart which shows the distribution of certain events over the hours of a day or the days of a week. For this, I can create the following queries, which have the hours or weekdays respectively on the X-axis:

```
select count(*) as count
from stream 
where ...
group by DatePart(@Timestamp, 'hour', OffsetIn('Europe/Zurich', @Timestamp)) as hour
order by hour
```

```
select count(*) as count
from stream 
where ...
group by DatePart(@Timestamp, 'weekday', OffsetIn('Europe/Zurich', @Timestamp)) as weekday
order by weekday
```

However, if I want to display this in a bar chart on a dashboard, the X-axis of the bar chart is always set to the time (instead of the hour or the weekday respectively. Am I missing something, or is this not supported on dashboards?

Moreover, is it possible to set labels or the dimension items in my examples (e.g. to show "Sunday" instead of "0" in the chart legend)?

**Comment by nblumhardt** [comment, marked as answer]
Hi! Unfortunately this isn't supported on dashboards right now. Some improvements in that area are on our radar for the future.

Choosing the "Table" chart display style will at least get the information you're after on the screen, though it won't be charted.

Hope this helps,
Nick

**Comment by tobias-graf-p** [comment]
OK, thanks for the quick reply. Can you already tell when you expect this to be available and is there a way to get notified when this improvements are available? I would also have some feedback/input for improvements in the dashboard area ...

**Reply by nblumhardt** [reply]
You're welcome. There's no schedule for this right now, but following the release notes/feature summaries that we post at https://docs.datalust.co/docs/whats-new is the best way to catch up on what we've added to Seq every few months.

We're also in the process of adding a public feature roadmap to datalust.co, I'll loop back here and let you know once that's published.

Best regards,
Nick
